packages/ui/src/components/GalleryPost/GalleryPost.tsx

```diff
@@ -0,0 +1,222 @@
+import { IMAGE_REGEX } from '@tloncorp/shared/dist';
+import {
+  ContentReference as ContentReferenceType,
+  PostContent,
+} from '@tloncorp/shared/dist/api';
+import * as db from '@tloncorp/shared/dist/db';
+import {
+  Image as ImageBlock,
+  Link,
+  VerseBlock,
+  VerseInline,
+} from 'packages/shared/dist/urbit';
+import { useMemo } from 'react';
+import { Dimensions, ImageBackground } from 'react-native';
+
+import { LinearGradient, View } from '../../core';
+import ChatContent from '../ChatMessage/ChatContent';
+import ContentReference from '../ContentReference';
+import Pressable from '../Pressable';
+import GalleryAuthorRow from './GalleryAuthorRow';
+
+const HEIGHT_AND_WIDTH = (Dimensions.get('window').width - 46) / 2;
+
+export default function GalleryPost({
+  post,
+  onPress,
+  onLongPress,
+}: {
+  post: db.Post;
+  onPress?: () => void;
+  onLongPress?: (post: db.Post) => void;
+}) {
+  const content = useMemo(
```

Comment:
Some of the helpers here look familiar -- could they be pulled in from anything existing?

Reply:
Will take a look.